instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpStatusCodeConverter.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter.http;
+
+import static java.util.Collections.unmodifiableMap;
+
+import io.opentelemetry.api.trace.StatusCode;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+// https://github.com/open-telemetry/semantic-conventions/blob/v1.21.0/docs/http/http-spans.md#status
+enum HttpStatusCodeConverter {
+  SERVER {
+
+    @Override
+    StatusCode getSpanStatus(int responseStatusCode) {
+      if (responseStatusCode >= 100 && responseStatusCode < 500) {
+        return StatusCode.UNSET;
+      }
+
+      return StatusCode.ERROR;
+    }
+
+    @Nullable
+    @Override
+    String getErrorType(int responseStatusCode) {
+      return serverErrorTypes.get(responseStatusCode);
+    }
+  },
+  CLIENT {
+    @Override
+    StatusCode getSpanStatus(int responseStatusCode) {
+      if (responseStatusCode >= 100 && responseStatusCode < 400) {
+        return StatusCode.UNSET;
+      }
+
+      return StatusCode.ERROR;
+    }
+
+    @Nullable
+    @Override
+    String getErrorType(int responseStatusCode) {
+      return clientErrorTypes.get(responseStatusCode);
+    }
+  };
+
+  abstract StatusCode getSpanStatus(int responseStatusCode);
+
+  @Nullable
+  abstract String getErrorType(int responseStatusCode);
+
+  private static final Map<Integer, String> serverErrorTypes;
+  private static final Map<Integer, String> clientErrorTypes;
+
+  // https://en.wikipedia.org/wiki/List_of_HTTP_status_codes
+  static {
+    Map<Integer, String> serverPhrases = new HashMap<>();
+    serverPhrases.put(500, "Internal Server Error");
+    serverPhrases.put(501, "Not Implemented");
+    serverPhrases.put(502, "Bad Gateway");
+    serverPhrases.put(503, "Service Unavailable");
+    serverPhrases.put(504, "Gateway Timeout");
+    serverPhrases.put(505, "HTTP Version Not Supported");
+    serverPhrases.put(506, "Variant Also Negotiates");
+    serverPhrases.put(507, "Insufficient Storage");
+    serverPhrases.put(508, "Loop Detected");
+    serverPhrases.put(510, "Not Extended");
+    serverPhrases.put(511, "Network Authentication Required");
```

Comment:
we changed this late in the PR to use the status code directly as the `error.type`, check out https://github.com/open-telemetry/semantic-conventions/pull/205#discussion_r1318747110

Reply:
I'll send a semconv PR to make this more explicit, I can see how this is unclear:

> SHOULD be set to the string representation of the status code

Reply:
https://github.com/open-telemetry/semantic-conventions/pull/331

Reply:
Huh, I was still under the impression we're supposed to use the reason phrase 😅 
I'll fix that, thanks for letting me know